internal/collectors/pci.go

```diff
@@ -0,0 +1,52 @@
+package collectors
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+
+	"github.com/SUSE/connect-ng/internal/util"
+)
+
+// var InfoOnly bool // indicated that -i/--info was used
+const PCIChecksumFile = "chksum-pcidata.txt"
+const pciTag = "pci_data"
+
+type PCI struct {
+	UpdateDataIDs bool
+}
+
+func (pci PCI) run(arch string) (Result, error) {
+	util.Debug.Print("pci.UpdateDataIDs: ", pci.UpdateDataIDs)
+	output, err := util.Execute([]string{"lspci", "-s", ".0"}, nil)
+	// Some systems may not have lspci command, so
+	// if the execute fails,
+	// send nil to SCC to indicate that PCI data was unavailable.
+	if err != nil {
+		return Result{pciTag: nil}, err
+	}
+	stringOutput := string(output)
+	// get sha256 has a pci data
+	mkSha256 := sha256.New()
+	mkSha256.Write([]byte(output))
+	sha256 := mkSha256.Sum(nil)
+	hwIndex := hex.EncodeToString(sha256)
+
+	oldHwIndex := util.GetChecksum(PCIChecksumFile)
```

Comment:
I feel like all this code is just a duplicate to that of the lsmod one. Could we have a common function that deals with this?

Reply:
I will look at that.